frontend/src/concepts/pipelines/context/PipelinesContext.tsx

```diff
@@ -89,6 +92,8 @@ type UsePipelinesAPI = APIState & {
   namespace: string;
   /** The Project resource behind the namespace */
   project: ProjectKind;
+  /** The DSPipeline resource for the namespace */
+  pipelineNamespaceCR: DSPipelineKind | null;
```

Comment:
Nope, no exposing this please. Make a modal wrapper or something in this file to avoid this. The context & the internal objects do not get exposed.